### PR TITLE
Add warning log to clean up orphaned CRDs

### PR DESCRIPTION
**Pull request by maqiuyujoyce**
### Change description

Follow up on #2972 .

### Tests you have done

- [N/A] Run `make ready-pr` to ensure this PR is ready for review.
- [X] Perform necessary E2E testing for changed resources.

Verified that the error showed up in the log when orphaned CRD existed, and disappeared after the orphaned CRD was deleted.

### CODE REVIEW

File: cmd/recorder/main.go [hunk 1]
```diff
@@ -194,7 +194,9 @@ func doRecord(ctx context.Context, c client.Client, gvks []schema.GroupVersionKi
 
 func forEach(c client.Client, gvk schema.GroupVersionKind, listOptions *client.ListOptions, fn func(unstructured.Unstructured) error) error {
 	for ok := true; ok; ok = listOptions.Continue != "" {
-		if _, ok := opk8s.IgnoredCRDList[gvkToCRDName(gvk)]; ok {
+		crdName := gvkToCRDName(gvk)
+		if _, ok := opk8s.IgnoredCRDList[crdName]; ok {
+			logger.Info(fmt.Sprintf("Orphaned CRD %s found, you should delete it using 'kubectl delete crd %s'", crdName, crdName))
```
Comment (yuwenma) [inline cmd/recorder/main.go:199]: Logger.Error?

Comment (maqiuyujoyce) [reply cmd/recorder/main.go:199]: I think it is not an error to keep orphaned CRDs. If we use logger.Error() here, then the recorder will act the same as what reported in https://github.com/GoogleCloudPlatform/k8s-config-connector/issues/2874 except for a more actionable message.

Comment (yuwenma) [reply cmd/recorder/main.go:199]: >  then the recorder will act the same as what reported in #2874 except for a more actionable message.

Right, we expect the users to fix their stale CRD by giving clear guidance and not break their usage. 

Also suggest adding the `logger := klog.FromContext(ctx)`.

Comment (maqiuyujoyce) [reply cmd/recorder/main.go]: Done.

File: cmd/recorder/main.go [hunk 2]
```diff
@@ -192,9 +192,14 @@ func doRecord(ctx context.Context, c client.Client, gvks []schema.GroupVersionKi
...
-func forEach(c client.Client, gvk schema.GroupVersionKind, listOptions *client.ListOptions, fn func(unstructured.Unstructured) error) error {
+func forEach(ctx context.Context, c client.Client, gvk schema.GroupVersionKind, listOptions *client.ListOptions, fn func(unstructured.Unstructured) error) error {
+	logger := klog.FromContext(ctx)
 	for ok := true; ok; ok = listOptions.Continue != "" {
-		if _, ok := opk8s.IgnoredCRDList[gvkToCRDName(gvk)]; ok {
+		crdName := gvkToCRDName(gvk)
+		if _, ok := opk8s.IgnoredCRDList[crdName]; ok {
+			logger.Error(fmt.Errorf("unexpected CRD %s", crdName),
+				fmt.Sprintf("Orphaned CRD %s found, you should delete "+
+					"it using 'kubectl delete crd %s'", crdName, crdName))
```
Comment (yuwenma) [inline cmd/recorder/main.go:202]: two nits:
1. go log does not use capitalized message. "Orphaned" --> "orphaned" 
2. the logger is a structured message. so it is better to be
```go
logger.Error( <error>, "please run `kubectl delete crd <crd_name>` to delete the orphaned CRD", "crd", crdName)
```

Comment (maqiuyujoyce) [reply cmd/recorder/main.go]: Done.

**Pull request merged**